test/fenv/fenv_gentests.cc

```diff
@@ -0,0 +1,280 @@
+/*
+ * fenv_gentests.cc
+ *
+ * RISC-V ISA: RV32F, RV32D, RV64F, RV64D
+ *
+ * Copyright (C) 2017-2024 Tactical Computing Laboratories, LLC
+ * All Rights Reserved
+ * contact@tactcomplabs.com
+ *
+ * See LICENSE in the top level directory for licensing details
+ *
+ */
+
+#include "fenv_test.h"
+
+constexpr unsigned maxtests_per_file = 500;
```

Comment:
A certain maximum number of tests per source file is generated. All 4000+ tests could be in one file, but it would be slower to compile and run. I tried different numbers of tests per file, and `500` was around optimal in terms of test runtime.